apps/dashboard/src/components/pay/PayAnalytics/components/PaymentHistory.tsx

```diff
@@ -0,0 +1,288 @@
+import { format } from "date-fns";
+import { useState } from "react";
+import { PaginationButtons } from "../../../../@/components/pagination-buttons";
+import { CopyAddressButton } from "../../../../@/components/ui/CopyAddressButton";
+import { ScrollShadow } from "../../../../@/components/ui/ScrollShadow/ScrollShadow";
+import { Badge } from "../../../../@/components/ui/badge";
+import { Skeleton } from "../../../../@/components/ui/skeleton";
+import { cn } from "../../../../@/lib/utils";
+import {
+  type PayPurchasesData,
+  getPayPurchases,
+  usePayPurchases,
+} from "../hooks/usePayPurchases";
+import { ExportToCSVButton } from "./ExportToCSVButton";
+import {
+  CardHeading,
+  NoDataAvailable,
+  TableData,
+  TableHeading,
+  TableHeadingRow,
+} from "./common";
+
+type UIData = {
+  purchases: PayPurchasesData["purchases"];
+  pages: number;
+};
+
+const pageSize = 10;
+
+export function PaymentHistory(props: {
+  clientId: string;
+  from: Date;
+  to: Date;
+}) {
+  const [page, setPage] = useState(1);
+
+  const purchasesQuery = usePayPurchases({
+    clientId: props.clientId,
+    from: props.from,
+    to: props.to,
+    start: (page - 1) * pageSize,
+    count: pageSize,
+  });
+
+  function getUIData(): {
+    data?: UIData;
+    isLoading?: boolean;
+    isError?: boolean;
+  } {
+    if (purchasesQuery.isLoading) {
+      return { isLoading: true };
+    }
+    if (purchasesQuery.isError) {
+      return { isError: true };
+    }
+
+    const purchases = purchasesQuery.data.purchases;
+    const totalCount = purchasesQuery.data.count;
+
+    if (purchases.length === 0) {
+      return { isError: true };
+    }
+
+    return {
+      data: {
+        purchases,
+        pages: Math.ceil(totalCount / pageSize),
+      },
+    };
+  }
+
+  const uiData = getUIData();
```

Comment:
this feels odd? why not just do the computation in a useMemo instead of defining the function and calling it immediately? or if that is truly what is wanted then make it an IIFE?

Reply:
Don't really need a memo here, I can make it an IIFE sure

Reply:
yeah but then u could just do it in-line, right?

Reply:
or if you want to centralize the logic then I'd prefer you move the function outside of the component definition and you *pass* it inputs (that today are coming from the component body)